sshd-core/src/main/java/org/apache/sshd/server/forward/TcpipServerChannel.java

```diff
@@ -215,10 +215,12 @@ protected OpenFuture doInit(Buffer buffer) {
         }
 
         if (streaming == Streaming.Async) {
+            int channelId = getId();
             out = new BufferedIoOutputStream(
```

Comment:
Hmmm. IIRC we had a number of attempts to deal with this in SSHD-1070. Guillaume added code that tried essentially to achieve the same at a lower level in #166. How does this interact with this new mechanism?

Also, we should double-check that we don't re-introduce SSHD-1069. (Though I think that was specific to the forwarding case with Streaming.Async, i.e., exactly this code here, which is off by default.)

Reply:
As far as I can tell this code does not affect the issues you mentioned. IMO it is aligned with the purpose of PR#166 of keeping the in-flight data size under control.

>> How does this interact with this new mechanism?

I think it "plugs a hole" so to speak since the low-level mechanism is not aware of the concerns raised in the `BufferIoOutputStream` pending write requests mechanism.